### DISCUSSION TITLE
Equality Norm Constraints

### DISCUSSION THREAD

**Discussion by Fengcheng-Pei** [cvxpy/cvxpy#1975] [Q&A]
Hello,

I am confused with my constraints. I have a matrix variable with size [8,2] and two norm constraints cp.norm(V[:,0])==5 and cp.norm(V[:,1])==5. But there is a error said that :

DCPError: Problem does not follow DCP rules. Specifically:
The following constraints are not DCP:
Pnorm(var5258[0:8, 0], 2) == 5.0 , because the following subexpressions are not:
|--  Pnorm(var5258[0:8, 0], 2) == 5.0
power(Pnorm(var5258[0:8, 1], 2), 2.0) == 5.0 , because the following subexpressions are not:
|--  power(Pnorm(var5258[0:8, 1], 2), 2.0) == 5.0

I do not know why and could someone helps me?
Thanks a lot!

Pei

**Comment by PTNobel** [comment]
Hey Pei,

So, convex optimization only allows for two types of constraints: affine equality constraints $A x = b$ and convex inequality constraints $f(x) \leq c$. Since norms aren't linear, you can only write $\lVert x\rVert \leq 5$ rather than $\lVert x \rVert = 5$.

If you're curious to read more, Section 4.2 of [Convex Optimization by Boyd and Vandenberghe](https://web.stanford.edu/~boyd/cvxbook/) explains it in more detail!

Parth
